typing/typedtree.mli

```diff
@@ -104,10 +106,19 @@ and pattern_desc =
             [row_desc] = [Some _] when translating [Ppat_type _],
                          [None] otherwise.
          *)
-  | Tpat_lazy of pattern
-        (** lazy P *)
-  | Tpat_exception of pattern
+  | Tpat_lazy of computation_pattern
```

Comment:
Does this mean that we really allow exceptions under `lazy`, or is it just future work?

Reply:
This is left for future work, but the idea is to allow them there, yes (various code in parmatch and matching would have to be updated for this to be supported in practice). I could remove this structural change from the PR (it's not important to the reason why I made the PR), but most of the complexity in the PR comes from `type_pat` which wouldn't be simplified. I'm happy to remove if there is agreement that it is important for a first iteration.